### PR TITLE
Problem: local state-sync commands are not integrated

### PR DESCRIPTION
**Pull request by yihuang**
- integrate commands
- write integration test

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,7 +31,7 @@
 - [#950](https://github.com/crypto-org-chain/cronos/pull/950) Implement memiavl and integrate with state machine.
 - [#985](https://github.com/crypto-org-chain/cronos/pull/985) Fix versiondb verify command on older versions
 - [#998](https://github.com/crypto-org-chain/cronos/pull/998) Bump grocksdb to v1.7.16 and rocksdb to v7.10.2
-
+- [#]() Integrate local state-sync commands.
```
Comment (yihuang) [inline CHANGELOG.md:34]:
```suggestion
- [#1027](https://github.com/crypto-org-chain/cronos/pull/1027) Integrate local state-sync commands.
```

File: integration_tests/test_basic.py [hunk 2]
```diff
@@ -1,4 +1,7 @@
 import json
+import shutil
```
Comment (yihuang) [inline integration_tests/test_basic.py:2]:
```suggestion
```

File: integration_tests/cosmoscli.py [hunk 3]
```diff
@@ -1613,3 +1615,36 @@ def changeset_ingest_versiondb_sst(self, versiondb_dir, sst_dir, **kwargs):
...
+        return self.raw(
+            "snapshots", "restore", height, format, home=self.data_dir,
+        ).decode()
+
+    def bootstrap_state(self, height):
+        '''
+        bootstrap cometbft state for local state sync
+        '''
+        return self.raw(
+            'tendermint', 'bootstrap-state', height, home=self.data_dir,
```
Comment (mmsqe) [inline integration_tests/cosmoscli.py:1649]:
```suggestion
            'tendermint', 'bootstrap-state', height=height, home=self.data_dir,
```

**Review by mmsqe — COMMENTED** [review]
might need add rpc_servers as well

File: go.mod [hunk 4]
```diff
@@ -237,7 +237,7 @@ replace (
 
 	github.com/peggyjv/gravity-bridge/module/v2 => github.com/crypto-org-chain/gravity-bridge/module/v2 v2.0.1-0.20230126051749-d984b1562242
 	// use cometbft
-	github.com/tendermint/tendermint => github.com/cometbft/cometbft v0.34.27
+	github.com/tendermint/tendermint => github.com/cometbft/cometbft v0.34.28
```
Comment (mmsqe) [inline go.mod:240]: maybe use v0.34.28 for submodules as well

**Review by JayT106 — APPROVED** [review]

File: go.mod (hunk 4 above)
Comment (yihuang) [reply go.mod:240]: submodules can wait for a point release of cosmos-sdk I think.

Comment (mmsqe) [reply go.mod:240]: seems has behaviour change in statesync https://github.com/cometbft/cometbft/compare/v0.34.27...v0.34.28

Comment (yihuang) [reply go.mod:240]: yeah, statesync test case failed after upgrade, but I haven't find out where to blame.

Comment (mmsqe) [reply go.mod:240]: seems can pass when revert https://github.com/crypto-org-chain/ethermint/pull/242/files

Comment (yihuang) [reply go.mod:241]: yeah, goleveldb is the key 👍

File: integration_tests/test_basic.py [hunk 5]
```diff
@@ -242,6 +246,89 @@ def test_statesync(cronos):
...
+    # wait for the network to grow a little bit
+    cli0 = cronos.cosmos_cli(0)
+    wait_for_block(cli0, 6)
+
+    sync_info = cli0.status()["SyncInfo"]
+    height = int(sync_info["latest_block_height"])
+    cronos.supervisorctl("stop", "cronos_777-1-node0")
+    # print(cli0.list_snapshot())
+    tarball = cli0.data_dir / "snapshot.tar.gz"
+    cli0.export_snapshot(height)
```
Comment (mmsqe) [inline integration_tests/test_basic.py:271]: height for export seems flaky, do we need fallback height if a more recent snapshot already exists

Comment (yihuang) [reply integration_tests/test_basic.py:271]: I can make it check the `list_snapshot` first.

File: go.mod (hunk 4 above)
Comment (yihuang) [reply go.mod:241]: https://github.com/cosmos/cosmos-sdk/pull/14952

**Review by mmsqe — APPROVED** [review]

**Pull request merged**